contract.sol

```diff
@@ -0,0 +1,152 @@
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+contract StakingRewardSystem is Ownable, ReentrancyGuard, PausableUpgradeable, Initializable {
+
+    IERC20 public stakingToken;  // The token that users will stake
+
+    // Structure to represent each stake
+    struct Stake {
+        uint256 amount;  // Amount of tokens staked
+        uint256 startTime;  // Timestamp when the stake was created
+    }
+    
+    uint256 public minStakingPeriod;  // Minimum period required to avoid early withdrawal penalties
+    uint256 public earlyWithdrawalPenalty;  // Penalty percentage for early withdrawal (scaled to 1e18)
+    uint256 public rewardRate;  // Rate at which rewards are calculated (scaled to 1e18)
+    uint256 public accumulatedPenalties;  // Holds the total penalties accumulated (in token's smallest unit)
+
+    address[] public rewardTokens;  // List of tokens available as rewards
+
+    // Mapping to track each user's stakes based on the reward token they selected
+    mapping(address => mapping(address => Stake)) public stakes;
+    mapping(address => bool) public isRewardToken;  // Tracks if a token is a valid reward token
+
+    // Events for logging key actions in the contract
+    event Staked(address indexed user, uint256 amount, address indexed rewardToken);
+    event Withdrawn(address indexed user, uint256 amount, uint256 penalty, address indexed rewardToken);
+    event RewardClaimed(address indexed user, uint256 rewardAmount, address indexed rewardToken);
+    event PenaltiesWithdrawn(address indexed owner, uint256 amount);
+
+    // Custom errors for more efficient error handling
+    error InvalidAmount();  // Thrown when staking amount is 0 or negative
+    error InvalidRewardToken();  // Thrown when an invalid reward token is specified
+    error WithdrawAmountExceedsStake();  // Thrown when trying to withdraw more than staked
+    error NoStakeFound();  // Thrown when trying to claim reward without a stake
+    error CannotClaimRewardYet();  // Thrown when attempting to claim reward too early
+    error NoPenaltiesToWithdraw();  // Thrown when no penalties are available for withdrawal by the owner
+
+    // Initializer function to replace constructor for upgradeable contracts
+    function initialize(address _stakingToken, address[] memory _rewardTokens) public initializer {
+        __Pausable_init();  // Initialize pausable functionality
```

Comment:
You have made the contract Pausable Init but no pause, unpause functions?
Please refer to the launchpad contract I shared and take idea from there.